### ISSUE TITLE
Post-Adobe future : maintaining a new fork?

### ISSUE THREAD

**Issue by yhancik** [adobe/brackets#15333]
Forking the discussion in issue #15329 into one focused on the following question : is there anyone within the Brackets community who's willing to start a sustainable fork, and has the time, energy & skills to maintain it?

When I say "anyone", it's clearly not going to be a single person.

As users, are we willing to financially support that development (as much as we all can afford, of course), after years of using this Adobe-backed project for free?

(I'm just starting the discussion, not offering to start that fork)

**Comment by fakerybakery** [comment]
I made a fork but know nothing about how Brackets is made

**Comment by killerog** [comment]
I wouldn't mind paying for Brackets, definitely if I could get some support why brackets has trouble starting up sometimes when windows 10 first starts ;-)

Same as everyone else though, no idea how brackets is build currently.

**Comment by scratchieguy12345678** [comment]
I MIGHT mantain a fork.

**Comment by scratchieguy12345678** [comment]
I MIGHT mantain a fork.

**Comment by fakerybakery** [comment]
> I MIGHT maintain a fork. Brackets is my go-to editor, and I don't want it to disappear completely.

Thank you SOO much!!!

**Comment by scratchieguy12345678** [comment]
> > I MIGHT maintain a fork. Brackets is my go-to editor, and I don't want it to disappear completely.
> 
> Thank you SOO much!!!

https://github.com/scratchieguy12345678/bracketsfork

**Comment by scratchieguy12345678** [comment]
> > > I MIGHT maintain a fork. Brackets is my go-to editor, and I don't want it to disappear completely.
> > 
> > 
> > Thank you SOO much!!!
> 
> **EDIT:** Unfortunately, I've decided to actually not maintain a fork. It seems like a really big responsibility which I don't have time for. Also, looking through the code, I realized it would be really hard to continue on from what Adobe had written. I'm really sorry about that.



> > > I MIGHT maintain a fork. Brackets is my go-to editor, and I don't want it to disappear completely.
> > 
> > 
> > Thank you SOO much!!!
> 
> **EDIT:** Unfortunately, I've decided to actually not maintain a fork. It seems like a really big responsibility which I don't have time for. Also, looking through the code, I realized it would be really hard to continue on from what Adobe had written.

Sorry about that.

**Comment by moeenio** [comment]
One man can't continue the effort of 100s of contributors, backed by a large company.

**Comment by CreativeQuilt** [comment]
I have a fork Called Orion (Needs work with README and some other files): https://github.com/CreativeQuilt/Orion
GitHub page (Needs work): https://creativequilt.github.io/Orion/

**Comment by rafalkrupinski** [comment]
> One man can't continue the effort of 100s of contributors, backed by a large company.

Obviously, it needs a group of maintainers, but someone has to start

**Comment by Maniues** [comment]
I think it would be a good idea to submit brackets to various larger OSS communities like the Apache Software Foundation instead of trying to keep the project alive on your own.
However, if it doesn't then at least you need to make a copy of the repository and not a fork that doesn't handle issues and other features.

**Comment by fakerybakery** [comment]
@CreativeQuilt CreativeQuilt made something called Orion but now it's down from GitHub. @CreativeQuilt's whole account is down. If anyone makes one, please, fix the PHP error code null thing.

**Comment by Maniues** [comment]
@fakerybakery Can you tell what Orion is? How is this thread related?

**Comment by fakerybakery** [comment]
It was a fork that was going to be maintained but now I can't even find user CreativeQuilt. Maybe got banned? Deleted his account?

**Comment by Maniues** [comment]
@fakerybakery Did he participate in this thread? I can see that one of the quoted messages does not exist. I suspect he deleted the account himself, however I have no information. There is no such user on gitlab

**Comment by fakerybakery** [comment]
Yes

**Comment by fakerybakery** [comment]
After this is over I am going to delete all my messages so as not to clutter this thread. Ok?

**Comment by Maniues** [comment]
Yes, please remove these messages. I will remove mine

**Comment by fakerybakery** [comment]
It was at https://github.io/CreativeQuilt/Orion but there is a 404 page. CreativeQuilt isn't here now! That's strange!

**Comment by fakerybakery** [comment]
Had a conversation with someone here but deleted so as not to clutter up the thread.

**Comment by fakerybakery** [comment]
> I think it would be a good idea to submit brackets to various larger OSS communities like the Apache Software Foundation instead of trying to keep the project alive on your own.
> However, if it doesn't then at least you need to make a copy of the repository and not a fork that doesn't handle issues and other features.

Like the Apache Incubator? 
http://incubator.apache.org/

**Comment by Maniues** [comment]
@fakerybakery
Apache Incubator from what I can see is a project where the whole community of a given program wants to enter Apache. In this case, it would have to look more like handing over OpenOffice to ASF. I will write to them if there is any option to accept such a project.

**Comment by Maniues** [comment]
In this case, as mentioned before: one or more people MUST NOT continue with the project. If ASF takes over the project, we have a MINIMUM amount of work to continue the project.

**Comment by fakerybakery** [comment]
I am currently using Brackets (at this moment) and hope ASF will take it over. Please keep me posted if they respond to your request.

**Comment by Maniues** [comment]
I sent an e-mail to Apache Software Foundation (apache@apache.org). As soon as I get a reply I will write here about it. Maybe I will publish full e-mail. ASF is currently the only possible idea that will keep Brackets alive at the moment. In fact, ASF is the only (or one of the few) chances of keeping Brackets alive.
**Edit:**
Creating a fork outside any foundation is possible, but someone would have to realistically take care of it, gather many people who co-created Brackets, look for people who did Pull Request, create one organization, and as we can see - no one is ready for such a step, especially, that Brackets is based on CEF (Chromium Embedded Framework), not Electron.js which means that for many people it is more difficult to understand Brackets and deal with it alone or with a smaller group. Therefore, Brackets needs a foundation to organize these people, find them, encourage them to participate, adapt a new server for add-ons, and relocate some of them. You can do it yourself, but for the new Brackets to work in the same way as Adobe Brackets, it would take months, maybe a year.

**Comment by willnode** [comment]
So sad this unique editor will be abandoned. The live preview is very useful to me.

I created [a fork](https://github.com/willnode/brackets) and I've modified it a little so that it passes TravisCI build. Maybe I will tweak it a little over my free time to suit my needs. But I agree with @Maniues some orgs should have take over this project in order to get overall good support.

**Comment by moeenio** [comment]
> As soon as I get a reply I will write here about it

So that means no reply yet?

**Comment by Maniues** [comment]
> > As soon as I get a reply I will write here about it
> 
> So that means no reply yet?

@locness3 I haven't received an answer yet. Today I will write whether the case is pending. If I don't get a reply within a week, I will write here about an alternative idea

**Comment by PaulAtST** [comment]
That would be great if Apache could take over brackets. It would be a shame to see such a great tool go away.

**Comment by moeenio** [comment]
@Maniues did you read the stuff over at https://incubator.apache.org/ ?

**Comment by Maniues** [comment]
>The Apache Incubator provides services to projects which want to enter the Apache Software Foundation (ASF).

This excludes the participation of Apache Brackets. We don't want to join the project but get a fork from Apache

**Comment by Maniues** [comment]
I didn't get any reply from ASF. I can write to Mozilla Foundation or to The Document Foundation. What you think about that?

**Comment by flopezre** [comment]
Apache Brackets?

**Comment by Maniues** [comment]
Please give like for Mozilla Foundation or rocket 🚀 for the document foundation
I will write to Mozilla or TDF tomorrow

**Comment by moeenio** [comment]
I don't except the Document Foundation to be interested in this. However, Mozilla used Brackets as the basis for their now dead Thimble project. But I'm not sure they would maintain a "generic" fork of it.

**Comment by Maniues** [comment]
@locness3 You're right that TDF may not be interested in this. However, I am afraid that Mozilla will want to make a strong integration with Firefox, leaving other things behind. But it's better than nothing, so I will write to Mozilla Foundation

**Comment by bigrob1122** [comment]
What do u mean 


Sent from the all new AOL app for iOS


On Saturday, May 29, 2021, 2:11 PM, Maniues ***@***.***> wrote:




@locness3 You're right that TDF may not be interested in this. However, I am concerned that Mozilla will want to move towards a strong integration with Firefox, leaving other things behind. But it's better than nothing, so I will write to Mozilla Foundation

**Comment by yogo1212** [comment]
Hi :-)

TL;DR
https://github.com/brackets-cont/brackets
If you think you should be on the team, open an issue. Ideally, be someone who has contributed.


I'd say: This is a good opportunity to fork. Get away from the CLA.
It justs needs some organising.
Adobe was kind enough to place the emotional deadline in September.
De-facto there haven't been any commits since march.

The most that can be expected from this repo is to point to the fork in some way.
If a maintainer can put a link to the new repo in the description - wonderful.

I'm up for an adventure and willing to help organise a new project.
But i'm not really apt with HTTP/CSS so I will need at least some help for code review.

Here's a fork in a new organisation:
https://github.com/brackets-cont/brackets
The purpose of the organisation is to prevent Brackets to become a personal project. If you think, you belong in there, just open an issue in the repo!

These things have been or in the process of being done:
- ping a handful of contributors from this repo and ask them to become maintainers for code review
- activate issues in the project settings
- protect the master branch and require reviews for merging

I've started collecting next steps here:
https://github.com/brackets-cont/brackets/projects/1
Please, join in!
I won't be doing it alone ;-)

Let's hope there's some genuine interest to improve behind the complaining to Adobe.

regards!

**Comment by yogo1212** [comment]
@willnode i had the audacity of just inviting you.
would you create a PR into https://github.com/brackets-cont/brackets from your repo?

**Comment by willnode** [comment]
@yogo1212 will do it in a moment.

**Comment by moeenio** [comment]
Is it too much to ask for an admin to pin this thread?

**Comment by abose** [comment]
Hello everyone,

 Just want to give a heads up on the [Phoenix project](https://github.com/aicore/phoenix) to take brackets forward. 

Phoenix brings brackets to native browsers with full extension support. Phoenix is free and MIT Licensed; based on the latest Brackets core and work done by the Mozilla thimble Project. Even though we have had web migration discussions in the community since the first days of brackets, it only became really possible with significant advances in browser capabilities from 2020(native fs access APIs, shared workers for IPC between Phoenix instances, virtualized node fs APIs in the browser). We have built phoenix for the web leveraging these.  Phoenix will not have a native installable counterpart, unlike [brackets-cont](https://github.com/brackets-cont/brackets).

The builds are still in early alpha and can be tried out here: https://phoenix.core.ai/src/ 
![image](https://user-images.githubusercontent.com/5336369/131360203-3ba2508a-9c82-4369-a8b9-530889a4760f.png)


We are working with devs involved with Mozilla thimble to update libraries part of the thimble project to the latest web technologies.  Hopefully, we will be able to get native filesystem access working by September.

There are still a lot of essential features like [live view](https://github.com/aicore/phoenix/issues/29), [LSP support](https://github.com/aicore/phoenix/issues/39), [node compatibility layer](https://github.com/aicore/phoenix/issues/11) that are broken and are actively being worked on. 

[Requesting support from the brackets community for feedback, feature requests, and contributions.](https://phoenix.core.ai/)

Best,
Phoenix Team :)

https://github.com/adobe/brackets/issues/15373

**Comment by Maniues** [comment]
@abose Thank you for forking brackets. 
It would be nice if you provided a native version of Phoenix however.
Thanks to the @yogo1212 for forking brackets too.

**Comment by abose** [comment]
Thanks for the suggestion @Maniues . Phoenix should be able to be hosted in an electron environment as we have migrated it away from brackets shell dependencies. But it should be able to provide all features that a native version does(including working on local projects) without any installs and directly in any device/browser. Will evaluate if a native version is needed once the browser port is complete.

**Comment by Maniues** [comment]
@abose Thanks for replying. Development in browser can be uncomfortable for some people (including me), even more so if they do not program websites and web applications, but programs in Java, C # or C ++

**Comment by xypd** [comment]
I actually welcome the chance to keep brackets in a portable form. I'd rather not use electron unless really needed. With that said, I agree that unless you keep a browser for just coding in, I think it could open the dev environment to random typical web slow downs and crashes.

I'm exited that this seem to be at least one viable path forward. And think it's smart to use the work others have done already. Cheers!
